prowler/providers/kubernetes/services/rbac/rbac_minimize_csr_approval_access/rbac_minimize_csr_approval_access.py

```diff
@@ -0,0 +1,43 @@
+from prowler.lib.check.models import Check, Check_Report_Kubernetes
+from prowler.providers.kubernetes.services.rbac.rbac_client import rbac_client
+
+
+class rbac_minimize_csr_approval_access(Check):
+    def execute(self) -> Check_Report_Kubernetes:
+        findings = []
+        for crb in rbac_client.cluster_role_bindings:
+            for subject in crb.subjects:
+                if subject.kind in ["User", "Group"]:
+                    report = Check_Report_Kubernetes(self.metadata())
+                    report.namespace = "cluster-wide"
+                    report.resource_name = subject.name
+                    report.resource_id = subject.uid if hasattr(subject, "uid") else ""
+                    report.status = "PASS"
+                    report.status_extended = f"User or group '{subject.name}' does not have access to update the CSR approval sub-resource."
+                    findings.append(report)
+                    for cr in rbac_client.cluster_roles:
+                        if cr.metadata.name == crb.roleRef.name:
+                            if cr.rules:
+                                for rule in cr.rules:
+                                    if (
+                                        rule.resources
+                                        and "certificatesigningrequests/approval"
+                                        in rule.resources
+                                        and rule.verbs
+                                        and any(
+                                            verb in rule.verbs
+                                            for verb in ["update", "patch"]
+                                        )
+                                    ):
+                                        report.namespace = "cluster-wide"
+                                        report.resource_name = subject.name
+                                        report.resource_id = (
+                                            subject.uid
+                                            if hasattr(subject, "uid")
+                                            else ""
+                                        )
```

Comment:
```suggestion

```

Reply:
Include a break here since you don't need to loop if the condition is met.

Reply:
Done